protocols/identify/src/lib.rs

```diff
@@ -137,11 +145,12 @@ impl<T: Callback> IdentifyProtocol<T> {
                 .misbehave(&info.peer_id, Misbehavior::TooManyAddresses(listens.len()))
         } else {
             trace!("received listen addresses: {:?}", listens);
+            let debug = self.debug;
             let reachable_addrs = listens
                 .into_iter()
                 .filter(|addr| {
                     multiaddr_to_socketaddr(addr)
-                        .map(|socket_addr| is_reachable(socket_addr.ip()))
+                        .map(|socket_addr| debug || is_reachable(socket_addr.ip()))
```

Comment:
Better wrapper this logic `debug || is_reachable(socket_addr.ip())` in `self.is_reachable()`.

Reply:
Can't do this, borrow check gank you

Reply:
Fine, then.